### PR TITLE
Bump the yacc pointer.

### LINKED ISSUE #1801: yacc issue: index out of range
likely identical twin of https://github.com/youtube/vitess/issues/767

```go
	s := "SELECT(F(F(F(F(F(F(F" +
		"(F(F(F(F(F(F(F(F(F(F" +
		"(F(F(F(F(F(F(F(F(F(F" +
		"(F(F(F(F(F(F(F(F(F(F" +
		"(F(F(F(F(F(F(F(F(F(T" +
		"(F(F(F(F(F(F(F(F(F(F" +
		"(F(F(F(F(F(F(F(F(F(F" +
		"(F(F(F(F(F(F(F(F(F(F" +
		"(F(F(F(F(F(F(F(F(F(F" +
		"(F(F(F(F(F(F(F(F(F((" +
		"F(0"

	Parse(s)
```

test code in #1802

stack trace:

>
goroutine 1 [running]:
github.com/cockroachdb/cockroach/sql/parser.sqlParse(0x2208386ac0, 0x2083ba400, 0x0)
	/var/folders/yy/4q8rrssd27vdgbr59w9qbffr0000gn/T/go-fuzz-build896428593/src/github.com/cockroachdb/cockroach/sql/parser/yaccpar:207 +0x548df
github.com/cockroachdb/cockroach/sql/parser.Parse(0x2083b1380, 0xcb, 0x0, 0x0, 0x0, 0x0, 0x0)
	/var/folders/yy/4q8rrssd27vdgbr59w9qbffr0000gn/T/go-fuzz-build896428593/src/github.com/cockroachdb/cockroach/sql/parser/parse.go:50 +0x16a

### PR DESCRIPTION
**Pull request by petermattis**
Use the newer yacc to re-generate the sql grammar.

Fixes #1801.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -129,7 +129,7 @@ check:
 	  grep -vE '(declaration of err shadows|^vet: cannot process directory \.git)'
 	@echo "golint"
 	@! golint $(PKG) | \
-	  grep -vE '(\.pb\.go|embedded\.go|_string\.go|LastInsertId|sql\.y)' \
+	  grep -vE '(\.pb\.go|embedded\.go|_string\.go|LastInsertId|sql/parser/(yaccpar|sql\.y):)' \
```
Comment (tamird) [inline Makefile]: what is yaccpar?

**Comment by tamird** [comment]
LGTM

File: Makefile (hunk 1 above)
Comment (petermattis) [inline Makefile]: It is the "file" containing the generated parser code and is annotated in `sql.go` with:

```
//line yaccpar:1
```

**Comment by tschottdorf** [comment]
LGTM.
do we know that this reliably fixes this class of panic?

**Comment by petermattis** [comment]
@tschottdorf The code that caused the panic has seen changes in the updated yacc code. I didn't look at the logs, but I'd guess that the similar problem go-fuzz found in the vitess parser resulted in changes to yacc.

**Comment by tschottdorf** [comment]
the posts in the sister thread in vitess sounded like "I dunno but it seems to work now". See https://github.com/youtube/vitess/issues/767.

**Comment by petermattis** [comment]
I was referring to the the git logs for `yacc.go`. Brief inspection right now reveals that https://github.com/golang/go/commit/a034e478a7067db7ec0f3b549436bf6fbee3f4a0 is most likely the source of the fix as the line of code just after the 7 new ones in that commit is where the index out of bounds panic occurred.

**Pull request merged**